src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -94,9 +113,46 @@ const AddPermissionsTable = ({ selectedPermissions, setSelectedPermissions, ...p
       dispatch(fetchRole(baseRoleUuid));
     }
 
+    debounbcedGetApplicationOptions = debounce(
```

Comment:
You should use `useCallback` instead of assigning it in `useEffect` it should be slightly faster and more readable.